main_script.py

```diff
@@ -0,0 +1,229 @@
+        barn_start_auction_str,
+        barn_end_auction_str,
+    ) = get_auction_range(str(start_block), str(end_block))
+
+    print(
+        "Production auctions from "
+        + prod_start_auction_str
+        + " until "
+        + prod_end_auction_str
+        + ".\n"
+        + "Barn auctions from "
+        + barn_start_auction_str
+        + " until "
+        + barn_end_auction_str
+        + ".\n"
+    )
+
+    final_consistency_rewards_budget = CONSISTENCY_REWARDS_BUDGET
+    # computing solver performance rewards
+    performance_rewards_per_solver = {}
+    performance_spend = 0
+    performance_rewards = compute_solver_rewards(str(start_block), str(end_block))
+    for index, row in performance_rewards.iterrows():
+        amount = int(row["primary_reward_xdai"])
+        if amount > 0:
+            performance_spend += amount
+        performance_rewards_per_solver[row["solver"]] = amount
+
+    for solver in performance_rewards_per_solver:
+        if performance_rewards_per_solver[solver] > 0:
+            performance_rewards_per_solver[solver] = (
+                performance_rewards_per_solver[solver]
+                * PERFORMANCE_REWARDS_BUDGET
+                / performance_spend
+            )
+        else:
+            final_consistency_rewards_budget += (-1) * performance_rewards_per_solver[
+                solver
+            ]
+
+    # computing quote rewards
+    quote_rewards_per_solver = {}
+    total_quotes = 0
+    quote_rewards = compute_quote_rewards(str(start_block), str(end_block))
+    for index, row in quote_rewards.iterrows():
+        number = int(row["num_quotes"])
+        if number > 0:
+            total_quotes += number
+            quote_rewards_per_solver[row["solver"]] = number
+
+    for solver in quote_rewards_per_solver:
+        quote_rewards_per_solver[solver] = (
+            quote_rewards_per_solver[solver] * QUOTE_REWARDS_BUDGET / total_quotes
+        )
+
+    # computing consistency rewards
+    participation_data = execute_participation_rewards_helper(
+        str(start_block), str(end_block)
+    )
+    participation_per_solver = {}
+    total_participation_tokens = 0
+    for index, row in participation_data.iterrows():
+        ll = list(row["competition_info"])
+        num_solutions = len(ll)
+        i = num_solutions - 1
+        while i >= num_solutions - 3 and i >= 0:
+            solver = ll[i]["solverAddress"]
+            if solver in participation_per_solver:
+                participation_per_solver[solver] += 1.2
+            else:
+                participation_per_solver[solver] = 1.2
+            i = i - 1
+            total_participation_tokens += 1.2
+        while i >= 0:
+            solver = ll[i]["solverAddress"]
+            if solver in participation_per_solver:
+                participation_per_solver[solver] += 1
+            else:
+                participation_per_solver[solver] = 1
+            i = i - 1
+            total_participation_tokens += 1
```

Comment:
This might require a comment or should be moved to its own function, as it is a bit convoluted.

Or use a for loop instead of a while loop:
```suggestion
        for i in range(num_solutions):
            if i >= num_solutions - 3:
                token_delta = 1.2
            else:
                token_delta = 1.0
            solver = ll[i]["solverAddress"]
            if solver in participation_per_solver:
                participation_per_solver[solver] += token_delta
            else:
                participation_per_solver[solver] = token_delta
            total_participation_tokens += token_delta
```